tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -499,7 +499,7 @@ def test_sample_independent_misc_arguments() -> None:
         assert sampler.sample_independent(study, trial, "param-a", dist) != suggestion
 
     sampler = TPESampler(
-        weights=lambda i: np.asarray([i * 0.11 for i in range(7)]), n_startup_trials=5, seed=0
+        weights=lambda i: np.asarray([10 - j for j in range(i)]), n_startup_trials=5, seed=0
```

Comment:
Ditto.

Reply:
For this test case, we actually had a problem with the previous test code. To begin with, the weight function passed to TPESampler is a function that takes a non-negative integer n and outputs an `ndarray` of length n. However, the weight passed in the previous test code outputs an `ndarray` of length 7 regardless of the given n.

It seems that in the ParzenEstimator before the refactoring, 0 was assigned to the weight even if the length of the weight was inappropriate and insufficient. ([ref](https://github.com/optuna/optuna/blob/master/optuna/samplers/_tpe/parzen_estimator.py#L113))

Since there are not that many users playing with weights, I think the impact will be limited, but should we take them into account? Should we consider them?

Reply:
In commit https://github.com/optuna/optuna/pull/2618/commits/1f4d6a9b1256ada98a96e9900067e12939115298, I have modified the `calculate_weights` code so that it does not destroy the code of users who use inappropriate weight functions. This commit preserves backward compatibility.